tests/std/tests/P0355R7_calendars_and_time_zones_clocks/test.cpp

```diff
@@ -0,0 +1,431 @@
+
+    static_assert(is_eq(leap <=> leap_second{equal, true, 0s}));
+    static_assert(is_lt(leap <=> leap_second{larger, true, 0s}));
+    static_assert(is_gt(leap <=> leap_second{smaller, true, 0s}));
+    static_assert(is_lteq(leap <=> leap_second{larger, true, 0s}));
+    static_assert(is_gteq(leap <=> leap_second{smaller, true, 0s}));
+    static_assert(is_lteq(leap <=> leap_second{equal, true, 0s}));
+    static_assert(is_gteq(leap <=> leap_second{equal, true, 0s}));
+    static_assert(noexcept(leap <=> leap_second{equal, true, 0s}));
+#endif // __cpp_lib_concepts
+
+    static_assert(noexcept(leap.date()));
+    static_assert(noexcept(leap.value()));
+    static_assert(leap_second{sys_seconds{42s}, true, 0s}.date() == sys_seconds{42s});
+    static_assert(leap_second{sys_seconds{42s}, true, 0s}.value() == 1s);
+    static_assert(leap_second{sys_seconds{42s}, false, 0s}.value() == -1s);
+}
+
+constexpr bool operator==(const leap_second_info& lhs, const leap_second_info& rhs) {
+    return lhs.is_leap_second == rhs.is_leap_second && lhs.elapsed == rhs.elapsed;
+}
+
+template <class DurationRep>
+void test_leap_second_info(const leap_second& leap, seconds accum) {
+    const bool is_positive = (leap.value() == 1s);
+    // First UTC time when leap is counted, before insertion of a positive leap, after insertion of a negative one.
+    const utc_time<duration<DurationRep>> utc_leap{leap.date().time_since_epoch() + accum + (is_positive ? 0s : -1s)};
+
+    auto lsi = get_leap_second_info(utc_leap - 1s);
+    assert(lsi == (leap_second_info{false, accum}));
+
+    lsi = get_leap_second_info(utc_leap - 500ms);
+    assert(lsi == (leap_second_info{false, accum}));
+
+    accum += leap.value();
+    lsi = get_leap_second_info(utc_leap);
+    assert(lsi == (leap_second_info{is_positive, accum}));
+
+    lsi = get_leap_second_info(utc_leap + 500ms);
+    assert(lsi == (leap_second_info{is_positive, accum}));
+
+    lsi = get_leap_second_info(utc_leap + 1s);
+    assert(lsi == (leap_second_info{false, accum}));
+}
+
+template <class Duration>
+void test_utc_clock_to_sys(const leap_second& leap) {
+    sys_time<Duration> before_leap;
+    if constexpr (is_integral_v<typename Duration::rep>) {
+        before_leap = leap.date() - Duration{1};
+    } else {
+        before_leap =
+            sys_time<Duration>{Duration{nextafter(leap.date().time_since_epoch().count(), typename Duration::rep{0})}};
+    }
+
+    auto u = utc_clock::from_sys(before_leap); // just before leap second
+    assert(utc_clock::from_sys(utc_clock::to_sys(u)) == u);
+    if (leap.value() == 1s) {
+        u += Duration{1};
+        assert(utc_clock::to_sys(u) == before_leap); // during
+    } else {
+        assert(utc_clock::from_sys(utc_clock::to_sys(u)) == u);
+    }
+
+    u += 1s;
+    assert(utc_clock::from_sys(utc_clock::to_sys(u)) == u); // just after
+}
+
+template <class Duration>
+void test_file_clock_from_utc(const leap_second& leap) {

```

Comment:
I don't know if it belongs necessarily in this file (though I think it certainly could), but can we test that the `filesystem::file_time_type` has been updated as well?